packages/nimble-components/src/patterns/error/specs/checkbox-radio-error.md

```diff
@@ -0,0 +1,103 @@
+
+-   Red error text below the component
+    -   The error text does not wrap
+    -   It has an ellipsis if the text is longer than the control
+    -   The `title` of the text is the full error string
+-   Red error icon (`nimble-icon-exclamation-mark`) on the right side of the label with a 4px margin between the icon and the right edge of the control
+-   Control indicator (checkbox square or radio circle) and error icon will both be center-aligned with the first line of label text, both when the label text wraps and when it fits on a single line.
+-   Control indicator and label will both be interactable -- hovering over them will put the component in the "mouseover" state and clicking them will activate the control
+-   The error icon along with the space between the label and the error icon will not be interactable
+
+Below are some examples of how the components will look and behave with different configurations. The blue outline shows the bounds of the control, and the blue background indicates the interactable regions of the control.
+
+![](spec-images/checkbox-no_error,no_wrap.PNG)
+
+![](spec-images/checkbox-no_error,wrap.PNG)
+
+![](spec-images/checkbox-error,no_wrap.PNG)
+
+![](spec-images/checkbox-error,wrap.PNG)
+
+![](spec-images/checkbox-long_error.PNG)
+
+![](spec-images/radio-horizontal,no_error.PNG)
+
+![](spec-images/radio-horizontal,error.PNG)
+
+![](spec-images/radio-vertical,error,no_wrap.PNG)
+
+![](spec-images/radio-vertical,error,wrap.PNG)
+
+![](spec-images/radio-vertical,long_error.PNG)
+
+### API
+
+#### Checkbox
+
+The `nimble-checkbox` will be updated to implement the `ErrorPattern` interface. This means it will have two new attributes:
+
+-   `error-text` - string
+-   `error-visible` - boolean
+
+The `error-visible` boolean will control whether or not a red exclamation icon will be visible next to the checkbox label and whether or not the `error-text` is displayed below the checkbox label.
+
+#### Radio Group
+
+The `nimble-radio-group` will be updated to implement the `ErrorPattern` interface. This means it will have two new attributes:
+
+-   `error-text` - string
+-   `error-visible` - boolean
+
+The `error-visible` boolean will control whether or not a red exclamation icon will be visible next to each radio label and whether or not the `error-text` is displayed below all the radio controls.
+
+While `error-visible` will be a new public attribute on the `nimble-radio-group`, it will also be added as an internal attribute on the `nimble-radio`, which will be responsible for displaying the error icon next to its label when that attribute is set. The `nimble-radio-group` will be responsible for keeping the `error-visible` state of each child `nimble-radio` instance synchronized with its own `error-visible` state. It will do this by updating `error-visible` on each `nimble-radio` any time its own `error-visible` state changes or any time its slotted radios changes.
```

Comment:
> it will also be added as an internal attribute

Maybe a nit, but particularly if it's internal maybe we can be a property instead of an attribute. But I can easily imagine styling being based off the attribute so in  that case 100% internal attribute makes sense.

Reply:
I made it an attribute so that it can be styled off of, particularly using the [existing error pattern styles](https://github.com/ni/nimble/blob/main/packages/nimble-components/src/patterns/error/styles.ts) that expects an `error-visible` attribute to control the visibility of the error.